### PR TITLE
Add Transformation::transform_point/vector and inverse. Add From/Into…

### PR DESCRIPTION
**Pull request by PvdBerg1998**
… nalgebra::Projective2

Fixes https://github.com/hecrj/coffee/issues/76.

### CODE REVIEW

**Comment by PvdBerg1998** [comment]
Do not merge yet

File: src/graphics/transformation.rs [hunk 1]
```diff
@@ -33,35 +34,67 @@ impl Transformation {
...
+        Point::new(point.x, point.y)
+    }
+
+    /// Transforms the given vector by the inverse of this transformation.
+    pub fn inverse_transform_vector(self, vector: Vector) -> Vector {
+        let vector = self.0
+            .try_inverse()
+            .expect("Transformation matrix should only contain invertible operations")
+            .transform_vector(&Vector3::new(vector.x, vector.y, 0.0));
+        Vector::new(vector.x, vector.y)
```
Comment (hecrj) [inline src/graphics/transformation.rs]: These methods feel out of scope, for now at least. I think users can do this by themselves by turning the `Transformation` into a `nalgebra::Matrix4` with the brand new trait implementations.

We shouldn't try to reimplement/wrap a math library.

File: src/graphics/transformation.rs [hunk 2]
```diff
@@ -33,35 +34,67 @@ impl Transformation {
...
 
     /// Creates a non-uniform scale transformation.
     ///
     /// It allows you to scale each axis independently. You should rarely need
     /// this.
-    pub fn nonuniform_scale(x: f32, y: f32) -> Transformation {
-        Transformation(nalgebra::Matrix4::new_nonuniform_scaling(
-            &nalgebra::Vector3::new(x, y, 1.0),
-        ))
+    pub fn nonuniform_scale(scale: Vector) -> Transformation {
```
Comment (hecrj) [inline src/graphics/transformation.rs]: Nice!

File: src/graphics/transformation.rs (hunk 1 above)
Comment (PvdBerg1998) [reply src/graphics/transformation.rs]: As said on discord, I disagree, since it:

1. Requires knowledge about the underlying math library
2. Requires adding and removing a third dimension (at least until Transformation uses a 3x3 matrix internally)
